packages/celotool/src/e2e-tests/validator_handshake_tests.ts

```diff
@@ -0,0 +1,263 @@
+      // The proxied validator will not connect to the new proxy because of the
+      // incorrect enode, so at this point the proxy does not have any proof
+      // it is on behalf of the proxied validator.
+      await restartInstance(gethConfig, gethConfig.instances[proxyIndex])
+      val0ValEnodeTable = (await jsonRpc(val0Web3, 'istanbul_getValEnodeTable', [])).result
+      let val0ProxiedValEntryOld = val0ProxiedValEntry
+      val0ProxiedValEntry = getValEnodeTableEntry(val0ValEnodeTable, proxiedValAddress)
+      assert(
+        val0ProxiedValEntry,
+        'No entry in restarted validator 0 for proxied validator after restarting the proxy'
+      )
+
+      assert(
+        // @ts-ignore
+        val0ProxiedValEntryOld.enode === val0ProxiedValEntry.enode,
+        'Proxied validator enode entry in validator 0 has changed'
+      )
+
+      // Restart the proxied validator to connect to the new proxy
+      await restartInstance(gethConfig, gethConfig.instances[proxiedValIndex], true)
+
+      // Give the proxy 30 seconds to connect to validator 0.
+      // This extra amount of time is needed because the proxy will have already
+      // tried to connect to validator 0 at start up, but been rejected because
+      // the proxy was not connected to its proxied validator. This 30 seconds
+      // waits for the proxy to retry connecting to proxy 0.
+      await sleep(30000)
+
+      const val0Peers = await val0Admin.getPeers()
+      const proxyEnodeNoPort = getEnodeNoPort((await proxyAdmin.getNodeInfo()).enode)
+      assert(
+        isValidatorPeer(val0Peers, proxyEnodeNoPort),
+        'Proxy is not a validator peer of validator 0'
+      )
+
+      // Verify that the peering was initiated by the proxy and used the handshake
+      const proxyPeer = getPeer(val0Peers, proxyEnodeNoPort)
+      assert(proxyPeer, "Could not find proxy in validator 0's peers")
+      assert(proxyPeer.network.inbound, 'Proxy peer to validator 0 is not inbound')
+    })
+  })
+})
+
+function getInstances() {
+  const instances: GethInstanceConfig[] = _.range(VALIDATORS - PROXIED_VALIDATORS).map((i) => ({
+    name: `validator${i}`,
+    validating: true,
+    syncmode: 'full',
+    port: 30303 + 3 * i,
+    rpcport: 8545 + 2 * i,
+    lightserv: false,
+    setNodeKey: false,
+  }))
+
+  for (let i = VALIDATORS - PROXIED_VALIDATORS; i < VALIDATORS; i++) {
+    instances.push({
+      name: `validator${i}`,
+      validating: true,
+      syncmode: 'full',
+      port: 30303 + 3 * i,
+      rpcport: 8545 + 2 * i,
+      lightserv: false,
+      proxy: `proxy${i}`,
+      isProxied: true,
+      setNodeKey: false,
+    })
+    instances.push({
+      name: `proxy${i}`,
+      validating: false,
+      syncmode: 'full',
+      port: 30303 + 3 * i + 1,
+      proxyport: 30303 + 3 * i + 2,
+      rpcport: 8545 + 2 * i + 1,
+      lightserv: false,
+      isProxy: true,
+      setNodeKey: true,
```

Comment:
Why is setting the node key important here?

Reply:
Originally this was to ensure that the proxied validator knew the enode of the proxy, however I just removed this now that when starting up get it will always set the node key if the node is a proxy